include/queues.h

```diff
@@ -58,7 +58,7 @@ typedef struct st_queue {
 #define queue_set_max_at_top(queue, set_arg) \
   (queue)->max_at_top= set_arg ? -1 : 1
 #define queue_remove_top(queue_arg) queue_remove((queue_arg), queue_first_element(queue_arg))
-typedef int (*queue_compare)(void *,uchar *, uchar *);
+typedef int (*queue_compare)(const void *, const void *, const void *);
```

Comment:
same. also, why to do the same twice? QUEUE can do simply
```c
  queue_compare compare;
```
it doesn't need to repeat the full declaration